text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+## Alternatives for identity providers (IdP)
+
+Several alternatives are discussed [here](https://docs.google.com/document/d/1mXrVAkUA9dd4M7fa_AJC8mQ55YnYJ-DKsGq30lh0FvA/edit#):
+
+* Running an IdP on crates.io: This would involve that crates.io managed their own identity provider. Since crates.io already relies on GitHub for identity, this can be left as an option for other registries.
+* *Proposed* External IDP such as GitHub: Use what's already there, and there is already association of crate owners with GitHub identities.
+
+# Prior art
+[prior-art]: #prior-art
+
+The topic of signing crates have been brought up several times in the past:
+
+* https://github.com/rust-lang/crates.io/issues/75 - this issue raises the initial concern
+* https://github.com/sigstore/community/issues/25 - contains an attempt at sigstore support, but focusing on rustup because cargo maintainers being overloaded
+* https://github.com/rust-lang/cargo/issues/4768 - an attempt to provide a signed crates.io index
+* https://github.com/withoutboats/rfcs/pull/7 - a modification of the above issue using TUF to signing index commits
+
+[NPM](https://thenewstack.io/npm-to-adopt-sigstore-for-software-supply-chain-security/), [Maven](https://blog.sonatype.com/maven-central-and-sigstore) and PyPI are all in the process of adopting Sigstore. It's not uncommon for organizations to use multiple programming languages, so integrating with Sigstore means that they can reuse the same infrastructure for verifying and auditing their software.
+
+The Sigstore community has written a [document](https://docs.google.com/document/d/1mXrVAkUA9dd4M7fa_AJC8mQ55YnYJ-DKsGq30lh0FvA/edit#) with recommendations and alternative designs for projects integrating Sigstore with artifact repositories.
+
+## Articles and papers
+
+The [Sigstore Blog](https://blog.sigstore.dev/) contains a lot of articles related Sigstore. In particular, [this post](https://blog.sigstore.dev/signatus-ergo-securus-who-can-sign-what-with-tuf-and-sigstore-ea4d3d84b8b6) covers the problems faced by similar package systems like PyPI.
+
+The GitHub tema has published a few [blog posts](https://github.blog/2022-10-25-why-were-excited-about-the-sigstore-general-availability/) on Sigstore. 
+
+A [paper](https://dl.acm.org/doi/abs/10.1145/3548606.3560596) about Sigstore on ACM.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Opt-in or opt-out of signing and/or verification? Making it opt-out could increase the percentage of signed crates, but making it opt-in to start with seems like the safest choice.
```

Comment:
IMHO: if signing isn't seamless enough that it could be considered burdensome for existing workflows, then it feels like the entire RFC was very misleading.